src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +17,34 @@
 
     @Override
     public Country add(Country country) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(country);
+            transaction.commit();
+        } catch (IllegalArgumentException e) {
+            throw new DataProcessingException("The instance of Country is not an entity.", e);
+        } catch (IllegalStateException | PersistenceException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Cannot save the country: " + country, e);
```

Comment:
Enhance the exception message by including more details about the `Country` entity, such as its name or other identifying attributes, to improve debugging and traceability.